main.cc

```diff
@@ -115,13 +117,13 @@ void gameInstance() {
                 defaultSetup(&game, whitePieces, blackPieces);
             }
 
-            Observer whitePlayer = constructPlayer(&game, wPlayer, &whitePieces);
-            Observer blackPlayer = constructPlayer(&game, bPlayer, &blackPieces);
+            Observer * whitePlayer = constructPlayer(&game, wPlayer, &whitePieces);
```

Comment:
Doesn't this have the same issue of stack memory being used (by initializing a Human object in constructPlayer) and then going out of scope (as we exit constructPlayer)?